server/server.go

```diff
@@ -45,21 +72,23 @@ func attemptTcpConnection(port int, opts *options.Options) error {
 	logger := opts.Logger
 	logger.Infof("Attempting to connect to port %d via TCP...", port)
 
-	_, err := net.Dial("tcp", fmt.Sprintf("0.0.0.0:%d", port))
+	defaultTimeout := time.Second * 5
```

Comment:
Should this be less than the total configured timeout for all the health checks?